jgrapht-core/src/main/java/org/jgrapht/intervalgraph/interval/Interval.java

```diff
@@ -0,0 +1,65 @@
+package org.jgrapht.intervalgraph.interval;
+
+public class Interval<T extends Comparable<T>> implements Comparable<Interval<T>> {
+    private T start;
+    private T end;
+
+    public Interval(T start, T end)  {
+        this.start = start;
+        this.end = end;
+        if (!isValid())
+            throw new RuntimeException();
+    }
+
+    public T getStart() {
+        return start;
+    }
+    public T getEnd() {
+        return end;
+    }
+
+    public boolean isIntersecting(Interval<T> other) {
+        return other.getStart().compareTo(this.getEnd()) <= 0 || this.getStart().compareTo(other.getEnd()) <= 0;
+    }
+
+    public boolean contains(T point) {
+        boolean result = point.compareTo(getStart()) >= 0 && point.compareTo(getEnd()) <= 0;
```

Comment:
Maybe this is just me, but it is hard for me to understand all of those compareTo calls. Maybe it would be good to write a small wrapper isBefore(point, getStart()) or similar.